File: Source/STSL.SmartLocker.Utils.Domain/LockerLease.cs
using STSL.SmartLocker.Utils.Common.Data;

namespace STSL.SmartLocker.Utils.Domain;

public sealed class LockerLease : EntityBaseWithTenant, IUsesGuidId
{
    public Guid Id { get; set; }
    public DateTimeOffset? StartedAt { get; set; }
    public DateTimeOffset? EndedAt { get; set; }
    public LockerBankBehaviour LockerBankBehaviour { get; set; }
    public bool EndedByMasterCard { get; set; } = false;
    
    public Guid? CardCredentialId { get; set; }
    public Guid? CardHolderId { get; set; }
    public Guid? LockerId { get; set; }
    public Guid? LockId { get; set; }

    // navigation
    public CardCredential? CardCredential { get; set; }
    public CardHolder? CardHolder { get; set; }
    public Locker? Locker { get; set; }
    public Lock? Lock { get; set; }
}

File: Source/STSL.SmartLocker.Utils.WebClient/src/app/shared/models/locker-lease.ts
import { ICardCredentialDTO } from './card-credential'
import { ICardHolderDTO } from './card-holder'
import { DateTimeOffset, IUsesGuidId } from './common'
import { ILockDTO } from './lock'
import { ILockerDTO } from './locker'
import { LockerBankBehaviour } from './locker-bank'

export interface ILockerLeaseDTO extends IUsesGuidId {
  startedAt?: DateTimeOffset;
  endedAt?: DateTimeOffset;
  lockerBankBehaviour: LockerBankBehaviour ;
  endedByMasterCard: boolean;
  cardCredential?: ICardCredentialDTO;
  cardHolder?: ICardHolderDTO;
  locker?: ILockerDTO;
  lock?: ILockDTO;
}
